src/main/java/com/artipie/rpm/Primary.java

```diff
@@ -90,60 +92,59 @@ public void update(final String key, final Pkg pkg) throws IOException {
                 .attr("ver", pkg.tag(Header.HeaderTag.VERSION))
                 .attr("rel", pkg.tag(Header.HeaderTag.RELEASE))
                 .up()
-                .add("checksum")
-                .attr("type", "sha256")
-                .attr("pkgid", "YES")
-                .set(new Checksum(pkg.path()).sha())
-                .up()
-                .add("summary")
-                .set(pkg.tag(Header.HeaderTag.SUMMARY))
-                .up()
-                .add("description")
-                .set(pkg.tag(Header.HeaderTag.DESCRIPTION))
-                .up()
-                .add("packager")
-                .set(pkg.tag(Header.HeaderTag.PACKAGER))
-                .up()
-                .add("url")
-                .set(pkg.tag(Header.HeaderTag.URL))
-                .up()
-                .add("time")
-                .attr("file", pkg.num(Header.HeaderTag.FILEMTIMES))
-                .attr("build", pkg.num(Header.HeaderTag.BUILDTIME))
-                .up()
-                .add("size")
-                .attr("package", Files.size(pkg.path()))
-                .attr("installed", pkg.num(Header.HeaderTag.SIZE))
-                .attr("archive", pkg.num(Header.HeaderTag.ARCHIVESIZE))
-                .up()
-                .add("location")
-                .attr("href", key)
-                .up()
-                .add("format")
-                .add("rpm:license")
-                .set(pkg.tag(Header.HeaderTag.LICENSE))
-                .up()
-                .add("rpm:vendor")
-                .set(pkg.tag(Header.HeaderTag.VENDOR))
-                .up()
-                .add("rpm:group")
-                .set(pkg.tag(Header.HeaderTag.GROUP))
-                .up()
-                .add("rpm:buildhost")
-                .set(pkg.tag(Header.HeaderTag.BUILDHOST))
-                .up()
-                .add("rpm:sourcerpm")
-                .set(pkg.tag(Header.HeaderTag.SOURCERPM))
-                .up()
-                .add("rpm:header-range")
-                .attr("start", pkg.header().getStartPos())
-                .attr("end", pkg.header().getEndPos())
-                .up()
-                .append(rpmProvides(pkg))
-                .append(rpmRequires(pkg))
-                .append(files(pkg))
-                .up()
-        );
+        )
+            .zipWith(
+                new Checksum(pkg.path()).sha(),
+                (directives, checksum) ->
+                    directives
+                        .add("checksum")
+                        .attr("type", "sha256")
+                        .attr("pkgid", "YES")
+                        .set(checksum)
+                        .up()
+            )
+            .map(
+                directives ->
+                directives
+                    .add("summary").set(pkg.tag(Header.HeaderTag.SUMMARY)).up()
```

Comment:
@Sammers21 this block can be moved to initial `Single.just()`

Reply:
If we would do it, we would change the order of tags. It does matter.